src/prefect/futures.py

```diff
@@ -292,6 +292,59 @@ def result(
             ) from exc
 
 
+def as_completed(futures: List[PrefectFuture], timeout=None) -> Iterator[PrefectFuture]:
+    """
+    An iterator over the given futures that yields each as it completes.
+
+    Args:
+        fs: The sequence of Futures (possibly created by different Executors) to
+            iterate over.
+        timeout: The maximum number of seconds to wait. If None, then there
+            is no limit on the wait time.
+
+    Returns:
+        An iterator that yields the given Futures as they complete (finished or
+        cancelled). If any given Futures are duplicated, they will be returned
+        once.
+
+    Raises:
+        TimeoutError: If the entire result iterator could not be generated
+            before the given timeout.
+
+    Examples:
+        ```python
+        @task
+        def sleep_task(seconds):
+            sleep(seconds)
+            return 42
+
+        @flow
+        def flow():
+            futures = random_task.map(range(10))
+            for future in as_completed(futures):
+                print(future.result())
+        ```
+    """
+    futures = set(futures)
```

Comment:
this overshadows the `futures` input arg, can we call this `unique_futures` or something?